saleor/core/models.py

```diff
@@ -143,20 +147,47 @@ class Meta:
         abstract = True
 
 
+class EventPayloadManager(models.Manager["EventPayload"]):
+    @transaction.atomic
+    def create_with_payload_file(self, payload: str) -> "EventPayload":
+        obj = super().create()
+        obj.save_payload_file(payload)
+        return obj
+
+    @transaction.atomic
+    def bulk_create_with_payload_files(
+        self, objs: Iterable["EventPayload"], payloads=Iterable[str]
+    ) -> list["EventPayload"]:
+        created_objs = self.bulk_create(objs)
+        for obj, payload_data in zip(created_objs, payloads):
+            obj.save_payload_file(payload_data)
+        return created_objs
+
+
 class EventPayload(models.Model):
+    PAYLOADS_DIR = "payloads"
+
     payload = models.TextField(default="")
     payload_file = models.FileField(
-        storage=private_storage, upload_to="payloads", null=True
+        storage=private_storage, upload_to=PAYLOADS_DIR, null=True
     )
     created_at = models.DateTimeField(auto_now_add=True)
 
+    objects = EventPayloadManager()
+
     def get_payload(self):
         if self.payload_file:
             with self.payload_file.open("rt") as f:
                 payload = f.read()
                 return payload
         return self.payload
 
+    def save_payload_file(self, payload_data: str):
+        prefix = get_random_string(length=12)
+        file_name = f"{self.pk}.json"
```

Comment:
@fowczarek I've shortened file name to make sure that we won't hit 256 KB message size limit while deleting payloads in batches by 1000